### PR TITLE
Feature/send email 2.0

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: src/utils/subgraphQuery.ts [hunk 1]
```diff
@@ -83,3 +83,41 @@ export const getValidContact = async (
...
+      }
+    `;
+
+    const variables = {
+      requiredSchema: ['email:string'],
+      id: protectedData,
+    };
+
+    const protectedDataResultQuery: GraphQLResponse =
+      await graphQLClient.request(SchemaFilteredProtectedData, variables);
```
Comment (SeddikBellamine) [inline src/utils/subgraphQuery.ts:112]: it's better to refactor the code into seperate functions or constants, that way we have a readable and modular code
```suggestion
const getQuery = (requiredSchema: string[], id: string) => {
  return gql`
    query CheckProtectedDataValidity($requiredSchema: [String!]!, $id: String!) {
      protectedDatas(
        where: {
          transactionHash_not: "0x",
          schema_contains: $requiredSchema,
          id: $id
        }
      ) {
        id
      }
    }
  `;
};

const createVariables = (protectedData: string) => {
  return {
    requiredSchema: ['email:string'],
    id: protectedData,
  };
};
export const checkProtectedDataValidity = async (
  graphQLClient: GraphQLClient,
  protectedData: string
): Promise<boolean> => {
  try {
    const query = getQuery(['email:string'], protectedData);
    const variables = createVariables(protectedData);

    const result: GraphQLResponse = await graphQLClient.request(query, variables);
```

File: src/utils/subgraphQuery.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+export const getValidContact = async (
+  graphQLClient: GraphQLClient,
+  contacts: Contact[]
+): Promise<Contact[]> => {
+  try {
+    // Contacts addresses
+    const contactsAddresses = contacts.map((contact) => contact.address);
+
+    // Query protected data
+    const SchemaFilteredProtectedData = gql`
```
Comment (SeddikBellamine) [inline src/utils/subgraphQuery.ts:14]: rename SchemaFilteredProtectedData to schemaFilteredProtectedData

File: src/utils/subgraphQuery.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          }
+          skip: $start
+          first: $range
+          orderBy: creationTimestamp
+          orderDirection: desc
+        ) {
+          id
+        }
+      }
+    `;
```
Comment (SeddikBellamine) [inline src/utils/subgraphQuery.ts:35]: seperate the querry in an other function to have a modular clean code like this : 
```suggestion
const getProtectedDataQuery = (requiredSchema: string[], id: string[], start: number, range: number) => {
  return gql`
    query GetValidContacts(
      $requiredSchema: [String!]!
      $id: [String!]!
      $start: Int!
      $range: Int!
    ) {
      protectedDatas(
        where: {
          transactionHash_not: "0x"
          schema_contains: $requiredSchema
          id_in: $id
        }
        skip: $start
        first: $range
        orderBy: creationTimestamp
        orderDirection: desc
      ) {
        id
      }
    }
  `;
};

export const getValidContact = async (
  graphQLClient: GraphQLClient,
  contacts: Contact[]
): Promise<Contact[]> => {
  try {
    // Contacts addresses
    const contactsAddresses = contacts.map((contact) => contact.address);

    // Pagination
    let protectedDataList: ProtectedDataQuery[] = [];
    let start = 0;
    const range = 1000;
    let continuePagination = true;

    do {
      const protectedDataQuery = getProtectedDataQuery(['email:string'], contactsAddresses, start, range);

      const variables = {
        requiredSchema: ['email:string'],
        id: contactsAddresses,
        start,
        range,
      };

      const protectedDataResultQuery: GraphQLResponse = await graphQLClient.request(
        protectedDataQuery,
        variables
      );
```

**Review by SeddikBellamine — CHANGES_REQUESTED** [review]
Minor changes:
*Refactor code
*Variable names should respect camelcase

File: src/utils/subgraphQuery.ts [hunk 4]
```diff
@@ -83,3 +83,41 @@ export const getValidContact = async (
     );
   }
 };
+
+export const checkProtectedDataValidity = async (
+  graphQLClient: GraphQLClient,
+  protectedData: string
+): Promise<boolean> => {
+  try {
+    const SchemaFilteredProtectedData = gql`
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/utils/subgraphQuery.ts]: Commit: 770d40222c6a10d76f104577bb42803848b1f7de

File: src/utils/subgraphQuery.ts (hunk 3 above)
Comment (Le-Caignec) [reply src/utils/subgraphQuery.ts]: Commit: 95f8780d1bf1c6dc257a9c2e8de19d13bd71ceef

File: src/utils/subgraphQuery.ts (hunk 1 above)
Comment (Le-Caignec) [reply src/utils/subgraphQuery.ts]: Commit: 95f8780d1bf1c6dc257a9c2e8de19d13bd71ceef

**Review by SeddikBellamine — APPROVED** [review]
LGTM!

**Pull request merged**